plugin.video.puls4/default.py

```diff
@@ -0,0 +1,310 @@
+                    poster = "%s%s" % (base_url,video['formatOverviewItemImgVersions']['low'])
+                    fanart = "%s%s" % (base_url,video['formatOverviewItemImgVersions']['hi'])
+                    desc = "%s \n%s\n%s \n\n(%s)" % (name,video['announcement'].encode('UTF-8'),item['title'].encode('UTF-8'),item['date'].encode('UTF-8'))
+                    if video.has_key("isVideo"):
+                        isVideo = video_item["isVideo"]
+                    else:
+                        isVideo = False
+                    addDirectory(title,fanart,fanart,desc,id,"getShowByID",isVideo)
+                
+def parseJsonGridVideoContent(url):
+    html = common.fetchPage({'link': url})
+    data = json.loads(html.get("content"))   
+    if data.has_key("rows"):
+        for video in data['rows']:
+            for item in video["cols"]:
+                if item.has_key("content"):
+                    for video_item in item['content']:
+                        isVideo = False
+                        name = video_item['channel'].encode('UTF-8')
+                        if video_item.has_key("isVideo") and video_item["isVideo"]:
+                            isVideo = video_item["isVideo"]
+                            id = video_item['objectId']
+                        else:
+                            id = video_item['channelId']
+                        title = "%s - %s" % (name,video_item['title'].encode('UTF-8'))
+                        logo = video_item['previewLink']
+                        poster = video_item['previewLinkVersions']['low']
+                        fanart = video_item['previewLinkVersions']['hi']
+                        if video_item.has_key("description"):
+                            desc = video_item['description'].encode('UTF-8')
+                        else:
+                            desc = ""                        
+                        
+                        if video_item.has_key("date"):
+                            date = "(%s)" % video_item['date'].encode('UTF-8')
+                        else:
+                            date = ""
+                            
+                        subline = "%s\n%s\n\n%s" % (name,desc,date)
+                        addDirectory(title,fanart,fanart,subline,id,"getShowByID",isVideo)
+                
+def getJSONVideos(url):
+    html = common.fetchPage({'link': url})
+    data = json.loads(html.get("content"))   
+    if data.has_key("videos"):
+        for video in data['videos']:
+            image = video["picture"]["orig"].encode('UTF-8')
+            id = video['id']
+            desc = cleanText(video["description"].encode('UTF-8'))
+            duration = video["duration"]
+            date = video["broadcast_date"].encode('UTF-8')
+            time = video["broadcast_time"].encode('UTF-8')
+            utime = video["broadcast_datetime"]
+            
+            day = datetime.datetime.fromtimestamp(float(utime)).strftime('%d').lstrip('0')
```

Comment:
you should store the created date instead of creating it from scratch everytime.
something like 
```
date = datetime.datetime.fromtimestamp(float(utime))
day = date.strftime('%d').lstrip('0')
hour = date.strftime('%H').lstrip('0')
```
also be careful because if you `lstrip('0')` a `00` string (like it's possible to have for midnight) you'll get an empty string..

The same in other places where you format the date